crates/ruma-events-macros/src/event_enum.rs

```diff
@@ -430,12 +432,53 @@ fn expand_content_enum(
 
     let marker_trait_impls = marker_traits(kind, ruma_events);
 
+    let redacted_ident = kind.to_redacted_content_enum();
+    let redaction_variants = variants.iter().map(|v| v.ctor(&redacted_ident));
+    let redacted_content: Vec<_> = events
+        .iter()
+        .map(|ev| to_event_content_path(kind, ev, Some("Redacted"), ruma_events))
+        .collect();
+
+    let redacted_content_enum = if kind.is_state() || kind.is_message() {
+        quote! {
+            #( #attrs )*
+            #[derive(Clone, Debug, #serde::Serialize)]
+            #[serde(untagged)]
+            #[allow(clippy::large_enum_variant)]
+            pub enum #redacted_ident {
+                #(
+                    #[doc = #event_type_str]
+                    #variant_decls(#redacted_content),
+                )*
+                /// Content of a redacted event not defined by the Matrix specification.
+                Custom(#ruma_events::custom::RedactedCustomEventContent),
+            }
+
+            impl #ident {
+                /// Redacts `Self` given a `RoomVersionId`.
+                pub fn redact(
```

Comment:
You were a little to late and I've merged the trait changes 😅 

This only needs a super simple tweak though:

```suggestion
            impl #ruma_events::RedactContent for #ident {
                fn redact(
```